File: backend/app/schemas/question.py
from pydantic import BaseModel
from typing import Optional, List

from app.db.models.question import QuestionType
from app.schemas.option import OptionResponse, OptionCreate

class QuestionBase(BaseModel):
    text: str
    question_type: QuestionType
    points: Optional[int] = 1

class QuestionCreate(QuestionBase):
    options: List[OptionCreate] = []

class QuestionUpdate(QuestionBase):
    pass

class QuestionResponse(QuestionBase):
    id: int
    quiz_id: int
    options: List[OptionResponse] = []

    class Config:
        from_attributes = True  # orm_mode yerine from_attributes kullanılıyor

class QuestionWithOptions(QuestionResponse):
    options: List[OptionResponse] = []

class QuestionGenerateRequest(BaseModel):
    topic: str
    difficulty: str
    num_questions: int = 5

File: android/app/src/main/java/com/farukaydin/quizapp/data/models/Question.kt
package com.farukaydin.quizapp.data.models

import com.farukaydin.quizapp.data.models.OptionCreate

data class Question(
    val id: Int,
    val text: String,
    val options: List<OptionResponse>,
    val correctOption: Int,
    val quizId: Int
)

data class QuestionCreate(
    val text: String,
    val question_type: String,
    val points: Int = 1,
    val options: List<OptionCreate>
) 